core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4Lob.java

```diff
@@ -91,23 +92,49 @@ public void read() throws SQLException {
 		}
 
 		T4Connection t4connection = this.connection_.getServerHandle().getT4Connection();
-		LogicalByteArray wbuffer = ExtractLobMessage.marshal(ExtractLobMessage.LOB_EXTRACT_BUFFER, lobHandle_, 1, 0,
+		LogicalByteArray wbuffer = ExtractLobMessage.marshal(ExtractLobMessage.LOB_EXTRACT_LEN, lobHandle_, 1, 0,
 				connection_.ic_);
 
 		LogicalByteArray rbuffer = t4connection.getReadBuffer(TRANSPORT.SRVR_API_EXTRACTLOB, wbuffer);
 		ExtractLobReply reply = new ExtractLobReply(rbuffer, connection_.ic_);
-		data_ = reply.lobDataValue;
+		length = reply.lobLength;
```

Comment:
Is this length the entire length of the lob  returned by the SQL layer ?

Reply:
Yes, it the entire length of the LOB.

Reply:
Ok that may be very large.  I thought we're supposed to return each chunk to application to consume ?  Will you implement that at next phase ?

Reply:
I have talked it with Selva. I think the logic is a little complex to support some APIs like setBytes(). I will add write down the design to discuss with you about the next phase.

Reply:
ok